File: system-test/sec-tests/java/Sec50ErrorLeak.java
// SEC-50: Error Handling Information Leak
// Test: TC-CR-054
package com.example;

import org.springframework.web.bind.annotation.*;

@RestController
public class ErrorController {

    @GetMapping("/api/data")
    public String getData() {
        try {
            return queryDatabase();
        } catch (Exception e) {
            return "Error: " + e.getMessage() + " — " + e.getStackTrace()[0];  // Leaks internal info
        }
    }
}

File: system-test/sec-tests/python/sec_50_error_leak.py
# SEC-50: Error Handling Information Leak
# Test: TC-CR-054
from flask import Flask, jsonify

app = Flask(__name__)

@app.route('/api/data')
def get_data():
    try:
        result = query_database()
        return jsonify(result)
    except Exception as e:
        return jsonify({"error": str(e)}), 500  # Leaks internal details
